src/pal/pal_posix.h

```diff
@@ -54,8 +54,12 @@ namespace snmalloc
     void notify_not_using(void* p, size_t size) noexcept
     {
       assert(is_aligned_block<OS_PAGE_SIZE>(p, size));
+#ifndef NDEBUG
```

Comment:
It would be nice to have this controlled by a separate flag that defaults to on in debug CMake configurations.